diff -Nur xf86-input-mouse-1.9.0.orig/src/bsd_mouse.c xf86-input-mouse-1.9.0/src/bsd_mouse.c
--- xf86-input-mouse-1.9.0.orig/src/bsd_mouse.c	2012-10-08 10:40:07.000000000 +0900
+++ xf86-input-mouse-1.9.0/src/bsd_mouse.c	2013-05-23 10:25:35.000000000 +0900
@@ -139,10 +139,10 @@
 {
     int i;
 
-    for (i = 0; internalNames[i]; i++)
+    for (i = 0; (sizeof(internalNames[i]) / sizeof(internalNames[0])) - 1; i++)
         if (xf86NameCmp(protocol, internalNames[i]) == 0)
             return TRUE;
-    for (i = 0; miscNames[i]; i++)
+    for (i = 0; (sizeof(miscNames[i]) / sizeof(miscNames[0])) - 1; i++)
         if (xf86NameCmp(protocol, miscNames[i]) == 0)
             return TRUE;
     return FALSE;
